coding/react.md

```diff
@@ -4,15 +4,33 @@ layout: base
 
 # React coding standards
 
-This document outlines our standards for writing React applications.
+This document outlines our standards for writing React applications and components.
 
-## Starter Project
+## Starting a new React Project
+When starting a new react application, we recommend [create-react-app](https://github.com/facebook/create-react-app) for bootstrapping the project. Avoid [ejecting](https://facebook.github.io/create-react-app/docs/available-scripts#npm-run-eject) as long as feasible to make updating dependencies easier.
 
-[canonical-webteam/react-starter](https://github.com/canonical-webteam/react-starter) provides a codified
-example of our standards for react projects (Prettier, AirBnB style).
+### Preferred libraries
```

Comment:
These suggestions are actually general to all JavaScript aren't they? It seems like they'd be better off sitting in their own document about general library choices for JavaScript, or general choices for front-end - similar to https://github.com/canonical-webteam/practices/blob/master/project-structure/server-side-frameworks.md.

Actually, as I'm writing this I realise that actually most of the recommendations in this document should be generalised as JavaScript guidelines - naming of component files and JSX files should probably sit alongside general advice for naming JavaScript files, that advice and advice on Redux could also be applicable in non-React apps. So what should probably happen is that any React-specific advice should form a section in a general document on JavaScript advice.

However, this is a much larger point, beyond the scope of this PR. So ignore it for now =)

I filed https://github.com/canonical-webteam/practices/issues/118 to track this point.

Reply:
> These suggestions are actually general to all JavaScript aren't they? 

Of the libraries mentioned, Jest is the only general js lib, the rest are all react specific.

> So what should probably happen is that any React-specific advice should form a section in a general document on JavaScript advice.

Sure, that sounds reasonable. The code formatting doc should probably be merged into the js doc as well.

Reply:
I thought y'all told me recently that Redux existed independently of React? I can see Enzyme and React-Router only work with React.

Reply:
> I thought y'all told me recently that Redux existed independently of React? I can see Enzyme and React-Router only work with React.

Sure, but as react is the only contemporary js framework we use in the company, as far as we're concerned `redux` is react specific.

Reply:
Redux is (in theory) independent library. But even their own docs/examples go very quickly from 'using Redux' to 'using Redux with React', that it's quite hard to see them apart, especially that Redux was created with React components in mind.

So as much as it can be used separately, Redux is quite strong part of React ecosystem. But, sure, if we want to have a recommended framework independent state management library - let's make it Redux, so we can use it consistently.